### DISCUSSION TITLE
Help with Brazilian Portuguese

### DISCUSSION THREAD

**Discussion by edivanteixeira** [neonbjb/tortoise-tts#129] [Q&A]
I would like to train voices but in another language, I looked in the documentation but it seems to me that the project only includes English, can anyone help me?

**Comment by neonbjb** [comment]
Happy to help anyone train this in a different language as long as they commit to open sourcing it. Step 1 would be gathering a large dataset of speech, at least 10,000 hours. If your language has a wav2vec2 speech recognition model available ([PR does](https://huggingface.co/jonatasgrosman/wav2vec2-large-xlsr-53-portuguese)), then the speech does not need to be transcribed.

Once you have this, reach out and I'll try and help you do further training

**Reply by williethewolf** [reply]
Can you elaborate on the requirements for those 10kh?

**Reply by neonbjb** [reply]
It's a loose figure. Tortoise was trained on ~50k hours of speech. Based on the model size and the number of epochs to convergence (2), I believe it underfit. 

I am sure you could train a model like this on less than 10k hours but you'll need to be very cognizant of overfitting. More importantly, if you have less than 10k hours, I would worry about the number of speakers you have in your dataset.

**Reply by williethewolf** [reply]
I know you mentioned in other posts about using Rogan's podcast. For training purposes, how much cleaning did you have to do to the samples? Did you just dump the full raw podcasts? did you cut them up into shorter clips?
10k is over a year of nonstop audio (almost 6 for 50K!), so I imagine that if you were to process those clips individually it would be at least 6 times that time assuming you work on that nonstop. And in that case, I'm assuming either the training is lenient with foreign artifacts like music or you just fed it pre-curated clips containing only voice.

I'm very interested in helping you make a Spanish model

**Reply by neonbjb** [reply]
Oh no way in hell did I do this manually. I built a bunch of scripts that do the splitting and filtering. I have a classifier that removes things like speakers talking over each other, background music, environmental noise, reverb, etc.

Scripts are here: https://github.com/neonbjb/DL-Art-School/blob/master/codes/scripts/audio/preparation/pipeline.py
Would be happy to share the model weights for the classifier, e-mail me for a link.

**Reply by kyjc3311** [reply]
> Happy to help anyone train this in a different language as long as they commit to open sourcing it. Step 1 would be gathering a large dataset of speech, at least 10,000 hours. If your language has a wav2vec2 speech recognition model available ([PR does](https://huggingface.co/jonatasgrosman/wav2vec2-large-xlsr-53-portuguese)), then the speech does not need to be transcribed.
> 
> Once you have this, reach out and I'll try and help you do further training

Hi, could you help me with training in Korean? Think I found the recognition model [KR](https://huggingface.co/kresnik/wav2vec2-large-xlsr-korean) or [here](https://huggingface.co/w11wo/wav2vec2-xls-r-300m-korean). What can I do for the next step?

**Reply by EdoanR** [reply]
Is this link helpful in some way? 
https://github.com/PaddlePaddle/PaddleOCR/issues/1048

**Reply by cobayashi7411** [reply]
> It's a loose figure. Tortoise was trained on ~50k hours of speech. Based on the model size and the number of epochs to convergence (2), I believe it underfit.
> 
> I am sure you could train a model like this on less than 10k hours but you'll need to be very cognizant of overfitting. More importantly, if you have less than 10k hours, I would worry about the number of speakers you have in your dataset.

I would like to try it. I would like to train it in my language. PLS.

**Reply by Erim32** [reply]
Hello @neonbjb, is it possible to update the requirement for 2023 version ? 
Just to have an idea, how many Go dataset represents for one language ?

Thk you

**Reply by fabioravila** [reply]
In our language, it is really difficult to find quality datasets. So creating a giant dataset is a very complicated task.
But if you get as much automation as you can, automating the entire process would make life a lot easier.

I'm working on a drag-n-drop style pipeline where we would simply place the audio or video sources and be able to separate the good from the bad and filter a good final collection.

I will send you an email requesting these weights to help with my pipeline.

**Reply by RahulBhalley** [reply]
Where is the training script???

**Reply by felipeduartedeveloper** [reply]
I'm also looking for support for Brazilian Portuguese

**Comment by dreamfaceprod** [comment]
> I would like to train voices but in another language, I looked in the documentation but it seems to me that the project only includes English, can anyone help me?

Hello, Edivan! 
do you have success? if yes, could we cooperate?

**Reply by josplit** [reply]
I would love if this worked in Spanish (in my test the subject actually does speak Spanish but with a weird French accent haha but it actually worked) is there any way we can help you with other languages? your work is incredible, thank you.

**Reply by djzigoh** [reply]
hi!!  any updates on this?  thanks!

**Comment by douglasescrito** [comment]
Eu também gostaria de ver essa ferramenta em portugues brasileiro.

**Comment by tiwale01** [comment]
Afternoon, any updates on this post? Was anything done ??

**Comment by alexandreespejo** [comment]
Any updates on this ?

**Reply by carlosedubarreto** [reply]
I'm not sure if it worth pursuing this work of making a brazilian one. I found another repository that already has some voices in pt-br that re great. its bark https://github.com/suno-ai/bark

**Reply by alexandreespejo** [reply]
Thanks a lot @carlosedubarreto, do you know if with bark is possible to train new voices models ?

**Reply by carlosedubarreto** [reply]
@alexandreespejo , the bark repo itself doesnt have, but the author suggest another one that you could use.

more info on this issue
https://github.com/suno-ai/bark/issues/37

**Reply by jjjota23** [reply]
achei o suno muito robótico... acho que vale a pena treinar o modelo acima
